analyses/bangladesh/trigger_development/read_in_data.py

```diff
@@ -0,0 +1,173 @@
+import sys
+import os
+from pathlib import Path
+
+import matplotlib as mpl
+from scipy.stats import norm, pearsonr
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+path_mod = f"{Path(os.path.dirname(os.path.realpath(''))).parents[1]}/"
+sys.path.append(path_mod)
+
+from src.indicators.flooding import glofas
+from src.bangladesh import get_glofas_data as ggd
+
+DATA_DIR = Path(os.environ["AA_DATA_DIR"])
+GLOFAS_DIR = DATA_DIR / "processed/bangladesh/GLOFAS_Data"
+STATION = "Bahadurabad_glofas"
+ffwc_dir = DATA_DIR / 'exploration/bangladesh/FFWC_Data'
+
+
+def get_glofas_reanalysis():
+    glofas_reanalysis = glofas.GlofasReanalysis(stations_lon_lat=ggd.FFWC_STATIONS)
+    da_glofas_reanalysis = glofas_reanalysis.read_processed_dataset(
+        country_name=ggd.COUNTRY_NAME, country_iso3=ggd.COUNTRY_ISO3
+    )[STATION]
+    return da_glofas_reanalysis
+
+
+def get_glofas_forecast():
+    glofas_forecast = glofas.GlofasForecast(
+        stations_lon_lat=ggd.FFWC_STATIONS, leadtime_hours=ggd.LEADTIME_HOURS
+    )
+    da_glofas_forecast_dict = {
+        leadtime_hour: glofas_forecast.read_processed_dataset(
+            country_name=ggd.COUNTRY_NAME,
+            country_iso3=ggd.COUNTRY_ISO3,
+            leadtime_hour=leadtime_hour,
+        )[STATION]
+        for leadtime_hour in ggd.LEADTIME_HOURS
+    }
+    da_glofas_forecast_dict = shift_dates(da_glofas_forecast_dict)
+    return convert_dict_to_da(da_glofas_forecast_dict)
+
+
+def get_glofas_reforecast():
+    glofas_reforecast = glofas.GlofasReforecast(
+        stations_lon_lat=ggd.FFWC_STATIONS, leadtime_hours=ggd.LEADTIME_HOURS
+    )
+    da_glofas_reforecast_dict = {
+        leadtime_hour: glofas_reforecast.read_processed_dataset(
+            country_name=ggd.COUNTRY_NAME,
+            country_iso3=ggd.COUNTRY_ISO3,
+            leadtime_hour=leadtime_hour,
+        )[STATION]
+        for leadtime_hour in ggd.LEADTIME_HOURS
+    }
+    da_glofas_reforecast_dict = interp_dates(shift_dates(da_glofas_reforecast_dict))
+    return convert_dict_to_da(da_glofas_reforecast_dict)
+
+
+def shift_dates(da_dict):
+    return {
+        leadtime_hour: da.assign_coords(
+            time=da.time.values + np.timedelta64(int(leadtime_hour / 24), "D")
+        )
+        for leadtime_hour, da in da_dict.items()
+    }
+
+
+def interp_dates(da_dict):
+    return {
+        leadtime_hour: da.interp(
+            time=pd.date_range(da.time.min().values, da.time.max().values),
+            method="linear",
```

Comment:
On looking back, my use of linear interpolation previously wasn't a very deliberate decision. Do you think it's worth considering another method, like cubic?

Reply:
No I think linear was a great choice! I don't think there's any compelling argument for using a more complex interpolation scheme, so I think it's safest to go with the simplest option. Also with other interpolation methods you can sometimes end up with wacky artifacts.